src/net/tcp/stream.rs

```diff
@@ -71,11 +72,17 @@ impl TcpStream {
     /// ```
     pub async fn connect<A: ToSocketAddrs>(addrs: A) -> io::Result<TcpStream> {
         let mut last_err = None;
+        let addrs = addrs
+            .to_socket_addrs()
+            .await
+            .context(|| String::from("could not resolve addresses"))?;
```

Comment:
It'd be great to know _which_ addresses couldn't be resolved. I feel this is one of the most common cases folks are left wondering about.